src/snmalloc/mem/corealloc.h

```diff
@@ -548,31 +546,52 @@ namespace snmalloc
      *
      * need_post will be set to true, if capacity is exceeded.
      */
+    template<typename Domesticator_queue>
     void handle_dealloc_remote(
       const PagemapEntry& entry,
-      CapPtr<void, capptr::bounds::Alloc> p,
-      bool& need_post)
+      capptr::Alloc<RemoteMessage> msg,
+      bool& need_post,
+      Domesticator_queue domesticate)
     {
       // TODO this needs to not double count stats
       // TODO this needs to not double revoke if using MTE
       // TODO thread capabilities?
 
       if (SNMALLOC_LIKELY(entry.get_remote() == public_state()))
       {
-        dealloc_local_object(p, entry);
-        return;
+        auto unreturned =
+          dealloc_local_objects_fast(entry, msg, entropy, domesticate);
+        if (SNMALLOC_UNLIKELY(unreturned.needs_work()))
+        {
+          dealloc_local_object_slow(msg.as_void(), entry);
```

Comment:
`dealloc_local_object_slow` might actually release a large object, but is otherwise a call to `dealloc_local_object_slower`, which, as you say, just updates metadata.  Maybe `dealloc_local_object_slow_meta` for the latter instead?

Reply:
That makes sense. It just felt in first inspection that the batch was added back in parts, rather than is all added back in one go, and then we might have to do several updates to get the meta data in the correct state.

Reply:
Yeah, you're right.  I've renamed the function (in a new commit at the bottom of the series) and will expand commentary on the later-introduced call site as part of the refactoring for your other comments. :)